netket/operator/_ising/base.py

```diff
@@ -67,9 +67,10 @@ def __init__(
         super().__init__(hilbert)
 
         if dtype is None:
-            dtype = jnp.promote_types(_dtype(h), _dtype(J))
-            # Fallback to float32 when float64 is disabled in JAX
-            dtype = jax.dtypes.canonicalize_dtype(dtype)
+            dtype = jnp.promote_types(float, _dtype(h))
+            dtype = jnp.promote_types(dtype, _dtype(J))
+        # Fallback to x32 when x64 is disabled in JAX
+        dtype = jax.dtypes.canonicalize_dtype(dtype)
```

Comment:
Actually I was also thinking about this, so let me do this now. We can use `jnp.result_type` rather than `jnp.promote_types`, which can take more than two arguments. What do you think?